Src/GoogleApis.Core/Apis/Requests/Parameters/ParameterUtils.cs

```diff
@@ -121,7 +121,17 @@ private static void IterateParameters(object request, Action<RequestParameterTyp
                 // Call action with the type name and value.
                 if (propertyType.IsValueType || value != null)
                 {
-                    action(attribute.Type, name, value);
+                    if (attribute.Type == RequestParameterType.UserDefiniedQueries)
+                    {
+                        foreach (var pair in (KeyValuePair<string, string>[])value)

```

Comment:
No problem, I'll wrap that in a type comparison. However for the logging (assuming you meant this), looking at the example of how it's set up in [AuthorizationCodeFlow](https://github.com/google/google-api-dotnet-client/blob/master/Src/GoogleApis.Auth/OAuth2/Flows/AuthorizationCodeFlow.cs#L44) and [MediaDownloader](https://github.com/google/google-api-dotnet-client/blob/master/Src/GoogleApis/Apis/%5BMedia%5D/Download/MediaDownloader.cs#L38) I would be unable to do this for ParameterUtils since the class is static.

If I try this:
```private static readonly ILogger Logger = ApplicationContext.Logger.ForType<ParameterUtils>();```

I get the error:
```Google.Apis.Requests.Parameters.ParameterUtils': static types cannot be used as type arguments```

Not sure how you'd want me to proceed on that one, so I'll hold off on that for now. Committing shortly.